zio-http/src/main/scala/zhttp/http/headers/HeaderChecks.scala

```diff
@@ -12,10 +13,11 @@ import zhttp.http.HeaderValues
  */
 trait HeaderChecks[+A] { self: HeaderExtension[A] with A =>
   final def hasContentType(value: CharSequence): Boolean = {
-    contentType.exists(h => {
-      val max = Math.min(value.length, h.length)
-      contentEqualsIgnoreCase(h.subSequence(0, max), value)
-    })
+    contentType
+      .flatMap(ct => Option(HttpUtil.getMimeType(ct)))
```

Comment:
This can throw is `ct == Null` – do you think we should handle that @tusharmath ?